projects/knora-ui/src/lib/viewer/values/decimal-value/decimal-value.component.spec.ts

```diff
@@ -0,0 +1,339 @@
+
+      commentInputNativeElement.value = 'this is a comment';
+
+      commentInputNativeElement.dispatchEvent(new Event('input'));
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.inputValueComponent.form.valid).toBeTruthy();
+
+      const updatedValue = testHostComponent.inputValueComponent.getUpdatedValue();
+
+      expect(updatedValue instanceof UpdateDecimalValue).toBeTruthy();
+
+      expect((updatedValue as UpdateDecimalValue).valueHasComment).toEqual('this is a comment');
+
+    });
+
+    it('should not return an invalid update value', () => {
+
+      testHostComponent.mode = 'update';
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.inputValueComponent.mode).toEqual('update');
+
+      expect(valueInputNativeElement.readOnly).toEqual(false);
+
+      expect(testHostComponent.inputValueComponent.form.valid).toBeFalsy();
+
+      expect(valueInputNativeElement.value).toEqual('1.5');
+
+      valueInputNativeElement.value = '.';
+
+      valueInputNativeElement.dispatchEvent(new Event('input'));
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.inputValueComponent.form.valid).toBeFalsy();
+
+      const updatedValue = testHostComponent.inputValueComponent.getUpdatedValue();
+
+      expect(updatedValue).toBeFalsy();
+
+    });
+
+    it('should restore the initially displayed value', () => {
+
+      testHostComponent.mode = 'update';
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.inputValueComponent.mode).toEqual('update');
+
+      expect(valueInputNativeElement.readOnly).toEqual(false);
+
+      expect(testHostComponent.inputValueComponent.form.valid).toBeFalsy();
+
+      expect(valueInputNativeElement.value).toEqual('1.5');
+
+      valueInputNativeElement.value = '40.09';
+
+      valueInputNativeElement.dispatchEvent(new Event('input'));
+
+      testHostFixture.detectChanges();
+
+      testHostComponent.inputValueComponent.resetFormControl();
+
+      expect(valueInputNativeElement.value).toEqual('1.5');
+
+      expect(testHostComponent.inputValueComponent.form.valid).toBeFalsy();
+
+    });
+
+    it('should set a new display value', () => {
+
+      const newDecimal = new ReadDecimalValue();
+
+      newDecimal.decimal = 40.09;
+      newDecimal.id = 'updatedId';
+
+      testHostComponent.displayInputVal = newDecimal;
+
+      testHostFixture.detectChanges();
+
+      expect(valueInputNativeElement.value).toEqual('40.09');
+
+      expect(testHostComponent.inputValueComponent.form.valid).toBeTruthy();
+
+    });
+
+    it('should unsubscribe when destroyed', () => {
+      expect(testHostComponent.inputValueComponent.valueChangesSubscription.closed).toBeFalsy();
+
+      testHostComponent.inputValueComponent.ngOnDestroy();
+
+      expect(testHostComponent.inputValueComponent.valueChangesSubscription.closed).toBeTruthy();
+    });
+  });
+
+  describe('create a decimaö value', () => {
```

Comment:
typo "decimaö"

Reply:
🤦🏻‍♂️I blame the swiss german keyboard. There are too many letters.